.gitlab-ci.yml

```diff
@@ -468,6 +468,31 @@ build-macos-subkey:
   tags:
     - osx
 
+build-rust-doc:
+  stage:                           build
+  <<:                              *docker-env
+  <<:                              *test-refs
+  needs:
+    - job:                         test-linux-stable
+      artifacts:                   false
+  variables:
+    <<:                            *default-vars
+    SKIP_WASM_BUILD:               1
+    RUSTDOCFLAGS:                  "--html-in-header $(pwd)/.maintain/rustdoc-header.html"
+  artifacts:
+    name:                          "${CI_JOB_NAME}_${CI_COMMIT_REF_NAME}-doc"
+    when:                          on_success
+    expire_in:                     7 days
+    paths:
+    - ./crate-docs/
+  script:
+    - rm -f ./crate-docs/index.html # use it as an indicator if the job succeeds
```

Comment:
why not removing all files here ? If it fails we don't use them AFAICT, if it succeed we don't need old files.